NPE in marketplace service when there are n*30-1 topics

When there is a multiple of 30 topics minus one in the marketplace (29, 59, etc.) the service will fail to retrieve the add-on list with this error:
```
16:15:44.640 [ERROR] [nity.CommunityMarketplaceAddonService] - Unable to retrieve marketplace add-ons
java.lang.NullPointerException: null
        at java.util.Arrays.stream(Arrays.java:5614) ~[?:?]
        at java.util.stream.Stream.of(Stream.java:1188) ~[?:?]
        at org.openhab.core.addon.marketplace.internal.community.CommunityMarketplaceAddonService.lambda$0(CommunityMarketplaceAddonService.java:191) ~[?:?]
```

This happens because Discourse lists topics by pages of 30, and includes a hint to the next page in responses:
```json
...
topic_list: {
  "more_topics_url": "/c/marketplace/69?page=2",
}
...
```

The service will retrieve the next page, until there is no such link anymore.
https://community.openhab.org/c/marketplace/69.json
https://community.openhab.org/c/marketplace/69.json?page=1
https://community.openhab.org/c/marketplace/69.json?page=2
...

However when there's n*30-1 topics the nth page will include a link, but it will not contain any topics and some of the usual fields will be missing:

```json
{"topic_list":{"can_create_topic":false,"per_page":30,"top_tags":["published","stable","binding","kar"],"topics":[]}}
```
In particular the response lacks a list of users and this makes this code throw the NPE:
https://github.com/openhab/openhab-core/blob/1d17166c79c79dfff5cbb4e96803a7d7d67ccf8f/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java#L191

--- a/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java
+++ b/bundles/org.openhab.core.addon.marketplace/src/main/java/org/openhab/core/addon/marketplace/internal/community/CommunityMarketplaceAddonService.java
@@ -177,7 +177,9 @@ public class CommunityMarketplaceAddonService implements AddonService {
 
                 try (Reader reader = new InputStreamReader(connection.getInputStream())) {
                     DiscourseCategoryResponseDTO parsed = gson.fromJson(reader, DiscourseCategoryResponseDTO.class);
-                    pages.add(parsed);
+                    if (parsed.topic_list.topics.length != 0) {
+                        pages.add(parsed);
+                    }
 
                     if (parsed.topic_list.more_topics_url != null) {
                         // Discourse URL for next page is wrong

